File: LeetCode/String/Easy/557. Reverse Words in a String III/main.py
"""
Steps to reverse words in a string
1) split the string of words into array of words
2) reverse each word personally
3) join them back
"""


class Solution:
    def reverseWords(self, s):
        s = s.split(" ")
        for i in range(len(s)):
            s[i] = s[i][::-1]
        s = " ".join(s)
        print(s)
        return s


obj = Solution()
obj.reverseWords("Hi Iam Prat")

File: LeetCode/String/Easy/557. Reverse Words in a String III/main.cpp
#include <iostream>
#include <algorithm>
using namespace std;
class Solution
{

public:
    string reverseWords(string s)
    {
        string ans = "";
        int l = 0;
        int r = 0;
        while (r <= s.length())
        {
            if (s[r] == ' ' || r == s.size())
            {
                reverse(s.begin() + l, s.begin() + r);
                l = r + 1;
            }
            r++;
        }
        return s;
    }
};

int main()
{
    Solution obj;
    string input = "Hi Iam Prathamesh";
    string ans = obj.reverseWords(input);
    cout << ans;
}